src/policy.c

```diff
@@ -863,23 +949,259 @@ pool_buildflavorcmp(Pool *pool, Solvable *s1, Solvable *s2)
     return 0;
   return pool_evrcmp_str(pool, f1 ? f1 : "" , f2 ? f2 : "", EVRCMP_COMPARE);
 }
-#endif
+
+void intersect_selection(Pool* pool, Id dep, Queue* prev)
+{
+  Queue tmp, res;
+  queue_init(&tmp);
+  // could use queue init buffer as optimization
+  queue_init(&res);
+
+  Id* pp, p;
+  pp = pool_whatprovides_ptr(pool, dep);
+  while ((p = *pp++) != 0)
+  {
+    queue_push(&tmp, p);
+  }
+
+  int i = 0, j = 0;
+  // set intersection, assuming sorted arrays
+  if (prev)
```

Comment:
this assumes that the Id's returned by pool_whatprovides are sorted... is that correct @mlschroe ?